src/core/Header/HeaderButton.tsx

```diff
@@ -98,6 +99,7 @@ export const HeaderButton = (props: HeaderButtonProps) => {
         {description && <div className='iui-description'>{description}</div>}
       </>
     ),
+    ...(!!menuItems && { menuItems }),
```

Comment:
Interesting..if I removed your changes and passed undefined in story, there is no error...
Anyway, something like `...{ menuItems }` would not work too?

Reply:
Maybe storybook does something about undefined values internally then.

`...{ menuItems }` would not change current behavior because it's already there (with `undefined` value).